### ISSUE TITLE
Advanced mode: use context to send to headless neovim?

### ISSUE THREAD

**Issue by BlueDrink9** [dbalatero/VimMode.spoon#35]
The v2 project looks absolutely awesome. Would you be willing to look into sending the context you can obtain to a headless neovim, and reading out the result, rather than possibly-imperfectly duplicating the correct behavior in lua? It seems like it would be a much more long-term sustainable approach, and would give the user the benefit of all the neovim tools, configurations and plugins?

**Comment by dbalatero** [comment]
Yes! This is my v3 plan haha.

I think I should at least proof of concept it next before moving forward
with any more imperfect implementations (and believe me, they are!)

**Comment by BlueDrink9** [comment]
Excellent, glad to hear. I'm working on my own imperfect, since X11 doesn't have any general-text emulators, but mine's more OS-generic so I envy being able to access the context.

**Comment by dbalatero** [comment]
Yeah totally. In testing on Mac, the latency to copy to the clipboard, read
it out, reset the cursors, etc is untenable, in case you go that route.

The v1 I did basically translates motions and operators into os-available
hot keys (alt+left to go back a word, etc) and fires them which gets you an
imperfect but pretty usable basic set of Vim commands!

**Comment by BlueDrink9** [comment]
I was considering clipboard, good to know about. Damn. That being said, if you only need it occasionally I'm hoping it isn't so bad. Mine will be even worse I imagine, because of the extra os-independent layer.

Yeah, I'm broadly going down the v1 route. I might look at how you've done your maps some time.

**Comment by dbalatero** [comment]
@BlueDrink9 I'm working on the neovim client version, and I have it passing keys pressed over RPC to Neovim: https://github.com/dbalatero/VimMode.spoon/compare/nvim

It's a mess right now but I'm just trying to get something basic working.

Have you looked into the accessibilty APIs on linux? This might provide you access to cursor position and line values: https://en.wikipedia.org/wiki/Assistive_Technology_Service_Provider_Interface

**Comment by alok** [comment]
@dbalatero would any of the updates in nvim since last year help with this? i'd be willing to sponsor this for up to $200.
